Aptoide/appcoins-wallet-android
# [WAL-54] Fingerprint remaining integration

**What does this PR do?**

Makes the following changes to the already existing fingerprint integration:

- Several code improvements
- Locked rotation on Authentication Activity to avoid a bad UX
- Action points described in ticket

**Database changed?**

No

**Where should the reviewer start?**

AuthenticationPromptActivity.kt

**How should this be manually tested?**

Check the 4 flows that require authentication (iap, entering wallet, remove wallet, disable authentication) with fingerprint
Disable device fingerprint and check if you are able to enter the wallet and the fingerprint settings is off and unable to set it on (future task)
On a device without fingerprint hardware check if the settings does not appear on settings

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/WAL-54


**Code Review Checklist**
